--- configuration.mustache (openapi-generator v7.25.0)
+++ configuration.mustache (tremendous)
@@ -1,6 +1,10 @@
 /* tslint:disable */
 {{>licenseInfo}}
 
+// Edited by Tremendous, lines 5-6, 92-93: Add custom User Agent with lib's version.
+const { version } = require("../package.json");
+const DefaultUserAgent = `Tremendous Node v${version}`;
+
 interface AWSv4Configuration {
   options?: {
     region?: string
@@ -85,15 +89,8 @@
         this.awsv4 = param.awsv4;
         this.basePath = param.basePath;
         this.serverIndex = param.serverIndex;
-        this.baseOptions = {
-            ...param.baseOptions,
-            headers: {
-                {{#httpUserAgent}}
-                'User-Agent': "{{httpUserAgent}}",
-                {{/httpUserAgent}}
-                ...param.baseOptions?.headers,
-            },
-        };
+        this.baseOptions = param.baseOptions || {};
+        this.baseOptions.headers = { "User-Agent": DefaultUserAgent };
         this.formDataCtor = param.formDataCtor;
     }
 
