--- index.mustache (openapi-generator v7.25.0)
+++ index.mustache (tremendous)
@@ -5,4 +5,5 @@
 
 export * from "./api{{importFileExtension}}";
 export * from "./configuration{{importFileExtension}}";
+export * from "./environments{{importFileExtension}}";
 {{#withSeparateModelsAndApi}}export * from "./{{tsModelPackage}}{{#importFileExtension}}/index{{importFileExtension}}{{/importFileExtension}}";{{/withSeparateModelsAndApi}}
